tofu/gcp/modules/elemental_nodes/main.tf

```diff
@@ -0,0 +1,21 @@
+resource "tls_private_key" "ssh" {
+  algorithm = "RSA"
+  rsa_bits  = 2048
+}
+
+# Elemental nodes
+module "elemental_nodes" {
+  source = "./../compute_instance"
+  instance_name = "${var.gcp_hostname_prefix}-elemental-node"
+  boot_image = var.boot_image
+  machine_type = var.machine_type
+  zone = var.zone
+  network = var.network
+  ssh_user = "ubuntu"
+  ssh_public_key = tls_private_key.ssh.public_key_openssh
```

Comment:
does GCP not require a key when creating an instance? I would argue that this could be part of the compute_instance instead, as we are pretty much always going to need ssh access to the node(s) created in the cloud.